Arm/State.lean

```diff
@@ -120,26 +120,29 @@ def PState.zero : PState :=
 
 @[ext]
 structure ArmState where
-  -- General-purpose registers: register 31 is the stack pointer.
-  gpr        : Store (BitVec 5) (BitVec 64)
-  -- SIMD/floating-point registers
-  sfp        : Store (BitVec 5) (BitVec 128)
-  -- Program Counter
-  pc         : BitVec 64
-  -- PState
-  pstate     : PState
-  -- Memory: maps 64-bit addresses to bytes
-  mem        : Store (BitVec 64) (BitVec 8)
-  -- Program: maps 64-bit addresses to 32-bit instructions.
-  -- Note that we have the following assumption baked into our machine model:
-  -- the program is always disjoint from the rest of the memory.
-  program    : Program
-
-  -- The error field is an artifact of this model; it is set to a
-  -- non-None value when some irrecoverable error is encountered
-  -- (e.g., an unimplemented instruction is hit). Any reasoning or
-  -- execution based off an erroneous state is invalid.
-  error      : StateError
+  /-- General-purpose registers: register 31 is the stack pointer. -/
+  private gpr        : Store (BitVec 5) (BitVec 64)
+  /-- SIMD/floating-point registers -/
+  private sfp        : Store (BitVec 5) (BitVec 128)
+  /-- Program Counter -/
+  private pc         : BitVec 64
+  /-- PState -/
+  private pstate     : PState
+  /-- Memory: maps 64-bit addresses to bytes -/
+  private mem        : Store (BitVec 64) (BitVec 8)
+  /--
+  Program: maps 64-bit addresses to 32-bit instructions.
+  Note that we have the following assumption baked into our machine model:
+  the program is always disjoint from the rest of the memory.
+  -/
+  program            : Program
```

Comment:
Out of scope for this PR, but it may be worth exploring if we ought to use `r` for accessing the `program` field as well, which'd solve the special treatment of this field here.

Reply:
Thanks!